lib/shellable.ts

```diff
@@ -153,6 +154,13 @@ export interface ShellableOptions {
    * @default No environment variables
    */
   readonly pipelineEnvironmentVars?: Record<string, string>;
+
+  /**
+   * The service role to assume while running the build
+   *
+   * @default A role will be created
```

Comment:
where is "a role will be created" happening?

Reply:
This happens in the codebuild `Project` class - it's the behavior that happens currently. See: https://github.com/aws/aws-cdk/blob/8557c40ce3efb16cb6c6f4677540d38853937166/packages/aws-cdk-lib/aws-codebuild/lib/project.ts#L1045

Reply:
Oh right. Okay, so the game plan is to send in our own serviceRole with similar permissions.

Reply:
Exactly. It would look something like this:
```ts
private addSigning() {
    const signingBucket = Bucket.fromBucketArn(this, 'SigningBucket', config.signingBucketArn);
    const signingLambda = Function.fromFunctionArn(this, 'SigningFunction', config.signingLambdaArn);
    const accessRole = Role.fromRoleArn(this, 'AccessRole', config.signingAccessRoleArn);
    const serviceRole = new Role(this, 'SeviceRole', {
      roleName: 'signing-codebuild-role',
      assumedBy: new ServicePrincipal('codebuild.amazonaws.com'),
    });
    serviceRole.addToPolicy(new PolicyStatement({
      actions: ['sts:AssumeRole'],
      resources: [accessRole.roleArn],
    }));

    this.pipeline.signNuGetWithSigner({
      signingBucket,
      signingLambda,
      accessRole,
      serviceRole,
    });
  }
```